lib/grandpa/message_handler.go

```diff
@@ -287,10 +309,30 @@ func (h *MessageHandler) verifyCommitMessageJustification(fm *CommitMessage) err
 }
 
 func (h *MessageHandler) verifyPreVoteJustification(msg *CatchUpResponse) (common.Hash, error) {
+	voters := make(map[ed25519.PublicKeyBytes]map[common.Hash]int)
+	eqVotesByHash := make(map[common.Hash]map[ed25519.PublicKeyBytes]struct{})
+
+	// identify equivocatory votes by hash
+	for _, just := range msg.PreVoteJustification {
+		voters[just.AuthorityID][just.Vote.Hash]++
+
+		if voters[just.AuthorityID][just.Vote.Hash] > 1 {
+			eqVotesByHash[just.Vote.Hash][just.AuthorityID] = struct{}{}
+		}
+	}
```

Comment:
can you do something like what you did in `verifyPreCommitJustification` here, so you can use the `getEquivocatoryVoters` function? eg.

```
	auths := make([]AuthData, len(msg.PreVoteJustification))
	for i, pcj := range msg.PreVoteJustification {
		auths[i] = AuthData{AuthorityID: pcj.AuthorityID}
	}

	eqvVoters := getEquivocatoryVoters(auths)
```